### PR TITLE
Petrov Maksim. Lab 1. Var 3.

### PR DESCRIPTION
**Pull request by soileks**

### CODE REVIEW

File: clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  CreateASTConsumer(CompilerInstance &Compiler,
+                    llvm::StringRef InFile) override {
+    return std::unique_ptr<ASTConsumer>(
+        new WarningDepricatedConsumer(Compiler));
+  }
+
+protected:
+  bool ParseArgs(const CompilerInstance &Compiler,
+                 const std::vector<std::string> &args) override {
+    return true;
```
Comment (m-ly4) [inline clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp:69]: Could you please add `help` argument and a test for it?

File: clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+using namespace clang;
+
+class WarningDepricatedVisitor
+    : public RecursiveASTVisitor<WarningDepricatedVisitor> {
+private:
+  ASTContext *Context;
+
+public:
+  explicit WarningDepricatedVisitor(ASTContext *Context) {this->Context = Context;}
```
Comment (m-ly4) [inline clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp]: Could you please pull last changes from main branch and fix clang-format and clang-tidy tests?

**Review by m-ly4 — CHANGES_REQUESTED** [review]

File: clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp (hunk 2 above)
Comment (soileks) [reply clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp]: Clang-format and clang-tidy have been fixed and the "help" argument has been added.

File: clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  CreateASTConsumer(CompilerInstance &Compiler,
+                    llvm::StringRef InFile) override {
+    return std::unique_ptr<ASTConsumer>(
+        new WarningDepricatedConsumer(Compiler));
+  }
+
+protected:
+  bool ParseArgs(const CompilerInstance &CI,
+                 const std::vector<std::string> &Args) override {
+    for (unsigned i = 0, e = Args.size(); i != e; ++i) {
```
Comment (soileks) [inline clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp]:
```suggestion
    for (unsigned I = 0, E = Args.size(); I != E; ++I) {
```

File: clang/labs/lab1/petrov_maksim/CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+    Support
+  )
+  clang_target_link_libraries(WarningDeprecatedPlugin PRIVATE
+    clangAST
+    clangBasic
+    clangFrontend
+    )
+endif()
+
+set(CLANG_TEST_DEPS "WarningDeprecatedPlugin" ${CLANG_TEST_DEPS} PARENT_SCOPE)
```
Comment (soileks) [inline clang/labs/lab1/petrov_maksim/CMakeLists.txt]:
```suggestion
set(CLANG_TEST_DEPS "WarningDeprecatedPlugin" ${CLANG_TEST_DEPS} PARENT_SCOPE)

```

File: clang/test/lab1/petrov_maksim/tests.cpp [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+// CHECK: warning: Function 'deprecatedFoo1' contains 'deprecated' in its name
+void deprecatedFoo1();
+
+// CHECK-NOT: warning: Function 'DeprecatedFoo2' contains 'deprecated' in its name
+void DeprecatedFoo2();
+
+struct St {
+// CHECK: warning: Function 'deprecatedFunc3' contains 'deprecated' in its name
+	void deprecatedFunc3() {}
```
Comment (soileks) [inline clang/test/lab1/petrov_maksim/tests.cpp]:
```suggestion
  void deprecatedFunc3() {}
```

File: clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                    llvm::StringRef InFile) override {
+    return std::unique_ptr<ASTConsumer>(
+        new WarningDepricatedConsumer(Compiler));
+  }
+
+protected:
+  bool ParseArgs(const CompilerInstance &CI,
+                 const std::vector<std::string> &Args) override {
+    for (unsigned I = 0, E = Args.size(); I != E; ++I) {
+      if (Args[i] == "help") {
```
Comment (soileks) [inline clang/labs/lab1/petrov_maksim/WarningDeprecatedPlugin.cpp]:
```suggestion
      if (Args[I] == "help") {
```

**Review by aobolensk — APPROVED** [review]

**Review by m-ly4 — APPROVED** [review]

**Pull request merged**